### ISSUE TITLE
2fA Issues in docker

### ISSUE THREAD

**Issue by grizzlybassist** [Zebradil/Gimme-iPhotos#40]
Running on a headless Debian 11 system with docker fully updated from Docker PPA.  Not sure what else to put here, just let me know, this will be my first issue submitted on github ever.

### Running command as follows:
```
docker run -it \
        -v /mnt/user/Uploads:/mnt/user/Uploads \
        -v ${PWD}/ryan.cfg:/app/ryan.cfg \
        zebradil/gimme-iphotos -c /app/ryan.cfg
```

### Contents of ryan.cfg:
```
[main]
username = ryan*******.com
password = **********
destination = /mnt/user/Uploads/icloud/ryan
remove = false
group = true
```

### Output and response:
```
Two-step authentication required.
Have you received authentication request on any of your devices? [y/N]: y
Please enter validation code: 464857
Failed to verify verification code, retry (Ctrl-C to cancel)
Please enter validation code: 464857
CRITICAL:root:Authentication required for Account. (421)
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gimme_iphotos/__init__.py", line 95, in main
    DownloaderApp(vars(args)).run()
  File "/usr/local/lib/python3.9/site-packages/gimme_iphotos/downloader.py", line 108, in run
    api = self.connect_to_icloud(config)
  File "/usr/local/lib/python3.9/site-packages/gimme_iphotos/downloader.py", line 155, in connect_to_icloud
    verified = api.validate_verification_code(device, code)
  File "/usr/local/lib/python3.9/site-packages/pyicloud/base.py", line 457, in validate_verification_code
    self.session.post(
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyicloud/base.py", line 128, in request
    return self.request(method, url, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyicloud/base.py", line 130, in request
    self._raise_error(response.status_code, response.reason)
  File "/usr/local/lib/python3.9/site-packages/pyicloud/base.py", line 186, in _raise_error
    raise api_error
pyicloud.exceptions.PyiCloudAPIResponseException: Authentication required for Account. (421)
```

**Comment by arickconley** [comment]
Same issue. Can't get docker to work.

**Comment by Zebradil** [comment]
Thanks for reporting!

I was able to reproduce this outside of docker too, will work on it.

**Comment by grizzlybassist** [comment]
I'm no python programmer, but I would love to help in any way I can, just let me know.

**Comment by Zebradil** [comment]
@grizzlybassist @arickconley are you able to perform 2FA via sms?

- You need to answer `n` to `Have you received authentication request on any of your devices? [y/N]`.
- Then you'll get a list of available devices for sms verification.
- Select a device with a number.
- Enter sms.

This seems to be an issue with the underlying library — [pyicloud](https://github.com/picklepete/pyicloud). Most likely Apple changed something in their API, I'm looking into that.

Another option, which worked for me is to run gimme-iphotos second time. In this case it doesn't require 2FA anymore, but then you have to have session cookies preserved somewhere.

Inside docker container, cookies are currently saved to `/tmp/pyicloud/root/youremail`. So, as a workaround, you can mount a directory from the host to `/tmp/pyicloud/root/` inside container.

I'm currently working on a feature to configure cookies destination directory, which will make this process more straightforward, but it's not ready yet.

**Comment by Zebradil** [comment]
Apparently, I was using incorrect 2fa code verification method. It was working before for both sms and 2fa code verification, but it seems like some time ago Apple stopped stopped providing backwards compatibility. Now two different endpoints must be used.

I created a PR, which should fix that, but before making a release, I would like to test it first.

Could you check if it works for you with this image: `zebradil/gimme-iphotos:pr-41` (when [this job](https://github.com/Zebradil/Gimme-iPhotos/runs/4218600517) is completed)?

**Comment by grizzlybassist** [comment]
Yes it is working.

**Comment by grizzlybassist** [comment]
I also tested mapping the cookie folder for running with no user interaction and it worked.  Thank you for helping with this so quickly.

**Comment by Zebradil** [comment]
Thanks for testing. You're welcome!

**Comment by Zebradil** [comment]
Released in [1.1.3](https://github.com/Zebradil/Gimme-iPhotos/releases/tag/1.1.3)
